apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx

```diff
@@ -141,20 +142,25 @@ const SwapForm = () => {
     }, idleTime);
   }, [idleState]);
 
-  const swapWebAppRedirection = useCallback(() => {
+  const swapWebAppRedirection = useCallback(async () => {
     const { to, from } = swapTransaction.swap;
     const transaction = swapTransaction.transaction;
     const { account: fromAccount, parentAccount: fromParentAccount } = from;
     const { account: toAccount, parentAccount: toParentAccount } = to;
-    const feesStrategy = transaction?.feesStrategy;
-    const rateId = exchangeRate?.rateId || "1234";
-    if (fromAccount && toAccount && feesStrategy) {
+    const { feesStrategy } = transaction || {};
+
+    const rateId = exchangeRate?.rateId || "12345";
+    if (fromAccount && toAccount) {
       const fromAccountId = accountToWalletAPIAccount(fromAccount, fromParentAccount)?.id;
       const toAccountId = accountToWalletAPIAccount(toAccount, toParentAccount)?.id;
       const fromMagnitude =
```

Comment:
I know this is not your code, but it could ease the readability to extract into a separate function the `fromAmount` value.

Reply:
Done